packages/browser-crawler/src/internals/browser-crawler.ts

```diff
@@ -716,13 +716,18 @@ export async function browserCrawlerEnqueueLinks({
 // eslint-disable-next-line @typescript-eslint/ban-types
 async function extractUrlsFromPage(page: { $$eval: Function }, selector: string, baseUrl?: string): Promise<string[]> {
     const urls = await page.$$eval(selector, (linkEls: HTMLLinkElement[]) => linkEls.map((link) => link.getAttribute('href')).filter((href) => !!href)) ?? [];
+    const [base] = await page.$$eval('base', (els: HTMLLinkElement[]) => els.map((el) => el.getAttribute('href')));
+
+    if (base) {
+        baseUrl = tryAbsoluteURL(base, baseUrl!);
```

Comment:
I don't know much about typescript, but the `baseUrl` is optional and here we're asserting it's non-null. Isn't that weird?